src/frontend/src/hooks/useExcelImport.ts

```diff
@@ -0,0 +1,119 @@
+import { useState, useCallback, useMemo, useEffect } from 'react';
+import { useAppContext } from '../appContext';
+import ExcelParseReponse, { ExcelHeader } from '../models/ExcelParseResponse';
+
+export type AutoGenerateColumn<T> = {
+    title: keyof T;
+    format: (columns: Column<T>[]) => {};
+};
+
+export type Column<T> = {
+    title: keyof T;
+    variations?: string[];
+    format?: (item: string) => {};
+};
+
+export type ExcelImportSettings<T> = {
+    columns: Column<T>[];
+    autoGenerateColumns?: AutoGenerateColumn<T>[];
+};
+
+type ColumnIndex<T> = {
+    [key in keyof T]: number;
+};
+
+const useExcelImport = <T>(excelImportSettings: ExcelImportSettings<T>) => {
+    const { apiClient } = useAppContext();
+
+    const [selectedFile, setSelectedFile] = useState<File | null>(null);
+    const [isProccessingFile, setIsProccessingFile] = useState<boolean>(false);
+    const [processedFile, setProcessedFile] = useState<T[] | null>(null);
+    const [error, setError] = useState<boolean>(false);
+    const { columns, autoGenerateColumns } = excelImportSettings;
+
+    useEffect(() => {
+        if (selectedFile === null) return;
+
+        processExcelFileAsync(selectedFile);
+    }, [selectedFile]);
+
+    const processExcelFileAsync = async (file: File) => {
+        setIsProccessingFile(true);
+        setError(false);
+
+        try {
+            const excelReponse = await apiClient.parseExcelFileAsync(file);
+
+            if (excelReponse) {
+                const formattedResponse = formatExcelReponse(excelReponse);
+                setProcessedFile(formattedResponse);
+            } else {
+                setProcessedFile(null);
+            }
+        } catch (e) {
+            setError(true);
```

Comment:
What errors might occur here? Will the backend throw anything "human readable" if it's unable to parse the excel-file?

Reply:
if a completly empty sheet is uploaded. The parses simply crashes. 
The error that coems back is not "human readable". 
Have implemented a generic error message for when an exception gets thrown. 

I have not been able to create an error with a file that has data in it.